### PR TITLE
Add a simple event counter to SNTExecutionController

### PR DESCRIPTION
**Pull request by pmarkowsky**
This PR adds a metric for execution events and updates tests to check for it. 

The singleton unfortunately makes tests order dependent.

Related-issue: #563

### CODE REVIEW

File: Source/santad/SNTExecutionController.m [hunk 1]
```diff
@@ -78,17 +80,45 @@ - (instancetype)initWithEventProvider:(id<SNTEventProvider>)eventProvider
     // This establishes the XPC connection between libsecurity and syspolicyd.
     // Not doing this causes a deadlock as establishing this link goes through xpcproxy.
     (void)[[MOLCodesignChecker alloc] initWithSelf];
+
+    SNTMetricSet *metricSet = [SNTMetricSet sharedInstance];
+    _events = [metricSet counterWithName:@"/santa/events"
+                              fieldNames:@[ @"action_response" ]
+                                helpText:@"Events processed by Santa with response"];
```
Comment (tburgin) [inline Source/santad/SNTExecutionController.m:87]: nit: "Events processed by Santa per response"

File: Source/santad/SNTExecutionControllerTest.m [hunk 2]
```diff
@@ -86,6 +87,24 @@ - (santa_vnode_id_t)getVnodeId {
   return (santa_vnode_id_t){.fsid = 1234, .fileid = 5678};
 }
 
+- (void)checkMetricCounters:(NSString *)expectedFieldValueName expected:(NSNumber *)expectedValue {
+  SNTMetricSet *metricSet = [SNTMetricSet sharedInstance];
+  NSDictionary *eventCounter = [metricSet export][@"metrics"][@"/santa/events"];
+  BOOL foundField;
+  for (NSDictionary *fieldValue in eventCounter[@"fields"][@"action_response"]) {
+    if ([expectedFieldValueName isEqualToString:fieldValue[@"value"]]) {
+      XCTAssertEqualObjects(expectedValue, fieldValue[@"data"], @"%@ counter not incremented",
```
Comment (tburgin) [inline Source/santad/SNTExecutionControllerTest.m:96]: nit: maybe counter not equal? For instance the counter could be incremented but not equal to the expected value.

**Review by tburgin — DISMISSED** [review]

File: Source/santad/SNTExecutionControllerTest.m (hunk 2 above)
Comment (pmarkowsky) [reply Source/santad/SNTExecutionControllerTest.m]: Yup good catch that was left over before refactoring this out.

File: Source/santad/SNTExecutionController.m (hunk 1 above)
Comment (pmarkowsky) [reply Source/santad/SNTExecutionController.m]: Yup. I've changed it.

**Review by tnek — APPROVED** [review]

**Pull request merged**